docker-compose.yaml

```diff
@@ -3,6 +3,32 @@ version: '3.4'
 services:
     db:
         image: postgres:11.2-alpine
+    s3:
+        image: minio/minio:RELEASE.2019-05-02T19-07-09Z
+        environment:
+            MINIO_ACCESS_KEY: &AWSKey AKIAIOSFODNN7EXAMPLE
+            MINIO_SECRET_KEY: &AWSSecret wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+            MINIO_REGION: &AWSRegion us-east-1
+        command: server /app
+        ports:
+            - 8081:9000
+    s3-bucket:
+        image: minio/mc:RELEASE.2019-05-01T23-27-44Z
+        environment:
+            AWS_ACCESS_KEY_ID: *AWSKey
+            AWS_SECRET_ACCESS_KEY: *AWSSecret
+            AWS_DEFAULT_REGION: *AWSRegion
+            S3_BUCKET: &S3Bucket bucket
+        entrypoint: sh
+        command: >
+            -ce '
+                while ! nc s3 9000; do sleep 1; done
+                mc config host add minio http://s3:9000 $${AWS_ACCESS_KEY_ID} $${AWS_SECRET_ACCESS_KEY}
+                mc mb minio/$${S3_BUCKET} --region $${AWS_DEFAULT_REGION}
+                mc policy download minio/$${S3_BUCKET}
```

Comment:
a bit arbitrary, real IAM policies are JSON documents authorizing various actions so what does this correspond to? We have our equivalent in https://github.com/libero/infrastructure/blob/master/tf/elife-style-content-adapter.tf#L21-L37 which is what I would use to implement the same behavior on an S3 bucket.
In addition to that, the application credentials will need a policy to add write access to the bucket.

Reply:
https://github.com/libero/content-store/blob/5a127bd17396ec3dedfc8f3ac0c80f6db486c95b/src/Workflow/MoveJatsAssets.php#L118-L125 sets access level on the object itself. But seems like you have to set something for it to work (https://docs.min.io/docs/minio-client-complete-guide.html#policy).

This is meant to mimic S3, doesn't have to have the same configuration. (Though it might be useful if a little obscure form of documentation.)

Reply:
(Related, the tests use an in-memory Flysystem.)